python/mxnet/contrib/onnx/_export/export_onnx.py

```diff
@@ -0,0 +1,270 @@
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+#  PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+#  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+#  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+#  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+#  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+#  OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+#  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+#  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+# coding: utf-8
+# pylint: disable=invalid-name,too-many-locals,no-self-use,too-many-arguments,
+# pylint: disable=maybe-no-member,too-many-nested-blocks
+"""MXNet to ONNX graph converter functions"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import json
+import numpy as np
+
+from .... import context
+from .... import ndarray as nd
+from .... import io
+from .... import module as mod
+
+
+class MXNetGraph(object):
+    """Class to convert MXNet to ONNX graph"""
+    registry_ = {}
+    input_output_maps_ = {}
+
+    def __init__(self):
+        # topologically sorted nodes
+        self.nodes = []
+        self.input_tensors = []
+        self.output_tensors = []
+
+    @staticmethod
+    def register(op_name):
+        """Register operator"""
+        def wrapper(func):
+            """Helper function to map functions"""
+            MXNetGraph.registry_[op_name] = func
+            return func
+
+        return wrapper
+
+    @staticmethod
+    def convert_layer(node, **kwargs):
+        """Convert MXNet layer to ONNX"""
+        op = str(node["op"])
+        if op not in MXNetGraph.registry_:
+            raise AttributeError("No conversion function registered for op type %s yet." % op)
+        convert_fun = MXNetGraph.registry_[op]
+        return convert_fun(node, **kwargs)
+
+    @staticmethod
+    def forward_pass(inputs, sym, arg_params, aux_params):
+        """ Do a forward pass based on the sym and params"""
+        data_names = [graph_input for graph_input in sym.list_inputs()
+                      if graph_input not in arg_params and graph_input not in aux_params
+                      and graph_input != 'softmax_label']
+
+        data_shapes = []
+        # Adding extra dimension of batch_size 1 if the batch_size is different for multiple inputs.
+        for idx, input_name in enumerate(data_names):
+            data_shapes.append((input_name, inputs[idx].shape))
+
+        # create module, passing cpu context
+        ctx = context.cpu()
+        test_mod = mod.Module(symbol=sym, data_names=data_names, context=ctx, label_names=None)
+        test_mod.bind(for_training=False, data_shapes=data_shapes, label_shapes=None)
```

Comment:
label_shapes may not always be None?

Reply:
True, but the motive of this function is to just get the shape of the output after forward pass.